<a href="https://colab.research.google.com/github/RafaelSastre/Datasets/blob/main/Presentacionintermedia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Ingresos de las lineas del metro 2012 - 2022

Rafael Sastré Félix A00830731

*   VARIABLES:






*   Dia: numerica (Indica en que dia se hizo el ingreso)


*   Año: numerica (indica en que año fue el ingreso)
*   Fecha: numerica( indica en que fecha fue el ingreso)


*   Tipo ingreso:categorica (Indica que significa el ingreso)
*   Linea: categorica (Indica en que linea se hizo el ingreso)

*   Ingreso: numerica (indica la cantidad de dinero que ingresó)














In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
path = '/content/presfinalestesi.xlsx'

In [3]:
df = pd.read_excel(path)
df

dia   año      fecha  tipo_ingreso linea   ingreso
0         1  2012 2012-01-01       Boletos     1  648825.0
1         1  2012 2012-01-01       Boletos     2  657120.0
2         1  2012 2012-01-01       Boletos     3  466803.0
3         1  2012 2012-01-01       Boletos     4   60120.0
4         1  2012 2012-01-01       Boletos     5  242280.0
...     ...   ...        ...           ...   ...       ...
143899    4  2022 2022-08-31  Validador QR     8       0.0
143900    4  2022 2022-08-31  Validador QR     9   14435.0
143901    4  2022 2022-08-31  Validador QR     A       0.0
143902    4  2022 2022-08-31  Validador QR     B       0.0
143903    4  2022 2022-08-31  Validador QR    12       0.0

[143904 rows x 6 columns]

In [4]:
df.columns

Index(['dia', 'año', 'fecha', 'tipo_ingreso', 'linea', 'ingreso'], dtype='object')

In [5]:

df

dia   año      fecha  tipo_ingreso linea   ingreso
0         1  2012 2012-01-01       Boletos     1  648825.0
1         1  2012 2012-01-01       Boletos     2  657120.0
2         1  2012 2012-01-01       Boletos     3  466803.0
3         1  2012 2012-01-01       Boletos     4   60120.0
4         1  2012 2012-01-01       Boletos     5  242280.0
...     ...   ...        ...           ...   ...       ...
143899    4  2022 2022-08-31  Validador QR     8       0.0
143900    4  2022 2022-08-31  Validador QR     9   14435.0
143901    4  2022 2022-08-31  Validador QR     A       0.0
143902    4  2022 2022-08-31  Validador QR     B       0.0
143903    4  2022 2022-08-31  Validador QR    12       0.0

[143904 rows x 6 columns]

Que linea tuvo mas ingresos en un dia 

In [6]:
df.groupby(['linea'])['ingreso'].max()

linea
1     7619910.0
2     8337195.0
3     7268655.0
4     1747095.0
5     2562030.0
6     1973325.0
7     3551025.0
8     5360265.0
9     3474285.0
12    3150373.0
A     2709630.0
B     5693535.0
Name: ingreso, dtype: float64

Se puede ver que la linea 2 en un dia pudo ingresar 8337195 pesos superando asi a las demas en un dia

Que lienea tuvo menos ingresos

In [7]:
df.groupby(['linea'])['ingreso'].sum()

linea
1     9.141621e+09
2     1.032538e+10
3     8.841817e+09
4     1.223978e+09
5     2.858167e+09
6     1.910599e+09
7     4.362643e+09
8     5.336679e+09
9     3.948715e+09
12    3.946654e+09
A     3.202921e+09
B     6.169709e+09
Name: ingreso, dtype: float64

Se pude ver que al sumar todos los ingresos que han tenido a lo largo de los años la linea 2 es la que mas a generado en estos 10 años generando 10,325,380,000 pesos 

Que tipo de ingresos aporta mas

In [8]:
df.groupby(['tipo_ingreso'])['ingreso'].sum()

tipo_ingreso
Boletos         2.971522e+10
Recargas        3.128483e+10
Tarjetas        2.656473e+08
Validador QR    3.186013e+06
Name: ingreso, dtype: float64

Se puede apreciar que a lo largo de estos 10 años lo que mas le ah ingresado dinero a las lineas del metro son las recargas de las tarjetas con 31,284,830,000 pesos

Que dia de la semana genera mas ingresos

In [9]:
df.groupby(['dia'])['ingreso'].sum()

dia
1    5.424760e+09
2    1.064377e+10
3    9.412358e+09
4    8.959551e+09
5    8.831437e+09
6    9.379066e+09
7    8.617932e+09
Name: ingreso, dtype: float64

Se ve que el dia lunes es el que mas dinero genera con diferencia sobre los otros dias de la semana ya que a generado 10,643,770,000 a lo largo de los 10 años

cual es el promedio de ingreso de cada linea

In [10]:
df.groupby(['linea'])['ingreso'].mean()

linea
1     762309.937553
2     861022.120038
3     737309.584125
4     102066.221338
5     238339.464516
6     159322.825239
7     363796.098885
8     445019.937516
9     329279.081355
12    329107.261687
A     267088.117535
B     514485.372723
Name: ingreso, dtype: float64

Se puede apreciar que la linea que mas aporta al metro es la linea 2 aportando 861,022.12 al dia en promedio




Cual es el promedio de ingreso de cada tipo de ingreso

In [11]:
df.groupby(['tipo_ingreso'])['ingreso'].mean()

tipo_ingreso
Boletos         635592.481883
Recargas        669165.512939
Tarjetas          5682.052426
Validador QR       873.358827
Name: ingreso, dtype: float64

Aqui se ve que las recargas aportan al dia en promedio 669,165.5129 pesos

que año tuvo mas ingresos 

In [12]:
df.groupby(['año'])['ingreso'].sum()

año
2012    4.463972e+09
2013    4.968039e+09
2014    6.763679e+09
2015    6.941626e+09
2016    7.143217e+09
2017    6.877182e+09
2018    6.835859e+09
2019    6.388239e+09
2020    3.999880e+09
2021    3.768411e+09
2022    3.118774e+09
Name: ingreso, dtype: float64

Al ver lo recaudado por cada año despues de  10 años el año que mas genero fue el año 2016 generando 7,143,217,000 pesos

cual es el promedio de los ingresos por dia


In [13]:
df.groupby(['dia'])['ingreso'].mean()

dia
1    263747.569521
2    517190.211080
3    457354.627519
4    435352.325271
5    430130.389667
6    456802.356757
7    419731.750927
Name: ingreso, dtype: float64

El dia que aporta mas dia a dia en promedio es el dia lunes que genera 517,190 pesos en promedio por dia 

cual es el promedio de ingresos de cada año

In [14]:
df.groupby(['año'])['ingreso'].mean()

año
2012    338795.675468
2013    378085.183240
2014    514739.663345
2015    528282.077413
2016    542138.507539
2017    523377.589338
2018    520232.815221
2019    486167.331126
2020    303573.157939
2021    271655.959153
2022    267384.643692
Name: ingreso, dtype: float64

Se puede ver que el promedio del dia de cada año el mayor es del 2016 generando alrededor de 542,138.50 pesos diarios 

Que año tuvo mas ingresos en un dia

In [15]:
df.groupby(['año'])['ingreso'].max()

año
2012    1877130.0
2013    8337195.0
2014    2544782.0
2015    2532217.0
2016    2566650.0
2017    2613643.0
2018    2836075.0
2019    2608248.0
2020    3497492.0
2021    2271572.0
2022    2619488.0
Name: ingreso, dtype: float64

El año que mas recaudo ingresos en un solo dia fue el año 2013 consiguendo ingresar 8,337,195 pesos en un dia